Add warnings for wrong values of model_confidence parameter (#8014)
RasaHQ/rasa: docs/docs/components.mdx

    set `weight_sparsity` to 1 as this would result in all kernel weights being 0, i.e. the model is not able
    to learn.

>   * `constrain_similarities`:

Reviewer: did we not describe this before anywhere? 🤔

Author: We did in the section where all configuration options of DIET are described. But this section contains options which we recommend users to tune. That's why I thought it would be good to have it here as well.

    This parameter when set to `True` applies a sigmoid cross entropy loss over all similarity terms.
    This helps in keeping similarities between input and negative labels to smaller values.
    This should help in better generalization of the model to real world test sets.

  * `model_confidence`:
    This parameter allows the user to configure how confidences are computed during inference. It can take two values:
    1. `softmax`: Confidences are in the range `[0, 1]` (old behaviour and current default). Computed similarities are normalized with the `softmax` activation function.

>     2. `inner`: Confidences are in the range `[0, 1]`. Computed similarities are normalized with a linear function.

Reviewer 2: let's use smth else then `inner`, and resort `inner` for raw inner

Author: `normalized_inner`?
I had thought of using `raw_inner`/`unnormalized_inner` for raw inner whenever we add that in.

Author: `normalized_inner` also doesn't sound right though, because softmax on top of inner is also some kind of normalization.

Author: We could use `relative_inner` for the linear normalized version of `inner` and `absolute_inner` for raw version. What do you think?

Reviewer 2: hm, not sure. `relative_inner` is also not very intuitive

Reviewer 2: @akelad do you have any thoughts on the name?

Reviewer: i don't think this name is *that* important tbh...

Reviewer: i have 0 opinion on it

Author: Changing it to `linear_norm` as discussed on [slack](https://rasa-hq.slack.com/archives/C8LEHMTAM/p1614249074146700?thread_ts=1614160186.107800&cid=C8LEHMTAM).
@akelad I'll ping you once the PR is ready for review again because I'll have to change the docs according to this renaming.

>     It is recommended to try using `linear_norm` as the value for `model_confidence` as it should be easier to tune thresholds for fallback then.

Reviewer: ```suggestion
    Please try using `linear_norm` as the value for `model_confidence`, this should make it easier to tune fallback thresholds for the FallbackClassifier
```
Can you also link to the fallback classifier?

    set `weight_sparsity` to 1 as this would result in all kernel weights being 0, i.e. the model is not able
    to learn.

  * `constrain_similarities`:
    This parameter when set to `True` applies a sigmoid cross entropy loss over all similarity terms.
    This helps in keeping similarities between input and negative labels to smaller values.
    This should help in better generalization of the model to real world test sets.

  * `model_confidence`:
    This parameter allows the user to configure how confidences are computed during inference. It can take two values:
    1. `softmax`: Confidences are in the range `[0, 1]` (old behaviour and current default). Computed similarities are normalized with the `softmax` activation function.
    2. `linear_norm`: Confidences are in the range `[0, 1]`. Computed dot product similarities are normalized with a linear function.

>     It is recommended to try using `linear_norm` as the value for `model_confidence` as it should be easier to tune thresholds for fallback then.

Reviewer: same comments here as above

>     It is recommended to try using `linear_norm` as the value for `model_confidence` as it should be easier to tune thresholds for fallback then.

Author: ```suggestion
    Please try using `linear_norm` as the value for `model_confidence`, this should make it easier to tune fallback thresholds for the [FallbackClassifier](./components.mdx#fallbackclassifier).
```
